src/lightning/fabric/strategies/xla_fsdp.py

```diff
@@ -140,6 +141,26 @@ def setup_module(self, module: Module) -> Module:
     def module_to_device(self, module: Module) -> None:
         pass
 
+    @contextmanager
+    def module_init_context(self, empty_init: Optional[bool] = None) -> Generator[None, None, None]:
+        # TODO: Use the meta device and reset parameters after https://github.com/pytorch/pytorch/issues/90465
+        # is resolved. For now, the module will get moved to the device in `setup_module`.
+        empty_init_context = _EmptyInit(enabled=bool(empty_init)) if _TORCH_GREATER_EQUAL_1_13 else nullcontext()
+        with empty_init_context, self.precision.init_context(), self.module_sharded_context():
+            yield
+
+    @contextmanager
+    def module_sharded_context(self) -> Generator:
+        from torch.distributed.fsdp.wrap import enable_wrap
```

Comment:
Does XLA FSDP also use `enable_wrap` from torch?

Reply:
I'm not sure here, `enable_wrap` does not exist in xla afaik

Reply:
Per our offline conversation, we will leave this as a no-op.

```
    @contextmanager
    def module_sharded_context(self) -> Generator:
        yield
```